Jenkinsfile.macos

```
@@ -1,175 +1,283 @@
+                      expression {
+                        return IS_PRO != true && FLAVOR == 'Electron';
+                      }
+                    }
+                    steps {
+                      // upload daily build redirects
+                      withCredentials([file(credentialsId: 'www-rstudio-org-pem', variable: 'wwwRstudioOrgPem')]) {
+                        sh 'docker/jenkins/publish-daily-binary.sh https://s3.amazonaws.com/rstudio-ide-build/${PRODUCT}/macos/${PACKAGE_FILE} ${RSTUDIO_ORG_PEM}'
+                      }
+                    }
+                  }
+                  
+                  stage("Sentry Upload") {
+                    environment {
+                      SENTRY_API_KEY = credentials('ide-sentry-api-key')
+                    }
+                    steps {
+                      // upload debug symbols to Sentry
+                      retry(5) {
+                        // timeout sentry in 15 minutes
+                        timeout(activity: true, time: 15) {
+                          // create new release on Sentry
+                          sh 'sentry-cli --auth-token ${SENTRY_API_KEY} releases --org rstudio --project ide-backend new ${RSTUDIO_RELEASE}'
+
+                          // upload Javascript source maps
+                          sh 'cd package/osx/build/gwt && sentry-cli --auth-token ${SENTRY_API_KEY} releases --org rstudio --project ide-backend files ${RSTUDIO_RELEASE} upload-sourcemaps --ext js --ext symbolMap --rewrite .'
+
+                          // associate commits
+                          sh 'sentry-cli --auth-token ${SENTRY_API_KEY} releases --org rstudio --project ide-backend set-commits --auto ${RSTUDIO_RELEASE}'
+
+                          // finalize release
+                          sh 'sentry-cli --auth-token ${SENTRY_API_KEY} releases --org rstudio --project ide-backend finalize ${RSTUDIO_RELEASE}'
+
+                          // upload C++ debug information
+                          sh 'cd package/osx/build/src/cpp && sentry-cli --auth-token ${SENTRY_API_KEY} upload-dif --org rstudio --project ide-backend -t dsym .'
+                        }
+                      }
+                    }
+                  }
+                  
+                  stage("Publish") {
+                    steps{
+                      // publish build to dailies page
+                      withCredentials([usernamePassword(credentialsId: 'github-rstudio-jenkins', usernameVariable: 'GITHUB_USERNAME', passwordVariable: 'GITHUB_PAT')]) {
+                        // publish the build
+                        sh "docker/jenkins/publish-build.sh --build ${PRODUCT}/macos --url https://s3.amazonaws.com/rstudio-ide-build/${PRODUCT}/macos/${PACKAGE_FILE} --pat ${GITHUB_PAT} --file package/osx/build/${PACKAGE_FILE} --version ${RSTUDIO_RELEASE}"
+                      }
```

The string uses insecure interpolation for a secret. Also we can use `VAR = credentials('username_psw')` to get `VAR_USR` and `VAR_PSW`. `VAR` itself ends up being `${VAR_USR}:${VAR_PSW}`
```suggestion
                  stage("Publish") {
                    environment {
                      GITHUB_LOGIN = credentials('github-rstudio-jenkins')
                    }
                    steps{
                      // publish build to dailies page
                      sh 'docker/jenkins/publish-build.sh --build ${PRODUCT}/macos --url https://s3.amazonaws.com/rstudio-ide-build/${PRODUCT}/macos/${PACKAGE_FILE} --pat ${GITHUB_LOGIN_PSW} --file package/osx/build/${PACKAGE_FILE} --version ${RSTUDIO_RELEASE}'
```

---

Done!